zkevm-circuits/src/keccak_circuit/keccak_padding_table.rs

```diff
@@ -0,0 +1,650 @@
+    var("DEGREE")
+        .unwrap_or_else(|_| "8".to_string())
+        .parse()
+        .expect("Cannot parse DEGREE env var as usize")
+}
+
+
+// [is_pads]          0 (data)   0 (data)  (pad) 1   (pad) 1  (pad) 1
+// [is_pad_starts]    0 (data)   0 (data)  (pad) 1   (pad) 0  (pad) 0
+// [is_pad_ends]      0 (data)   0 (data)  (pad) 0   (pad) 0  (pad) 1
+// [d_bytes]             79         106       128       0        1      [0]*CAPACITY//8
+// [d_lens]            base+1     base+2    base+2   base+2   base+2
+// [rlc] 
+
+#[derive(Debug, Clone)]
+#[allow(missing_docs)]
+pub struct PaddingCombinationsConfig<F> {
+    pub byte_col: TableColumn,
+    pub is_padding_col: TableColumn,
+    pub is_padding_start_col: TableColumn,
+    pub is_padding_end_col: TableColumn,
+    _marker: PhantomData<F>,
+}
+
+impl<F: Field> PaddingCombinationsConfig<F> {
+
+    pub(crate) fn configure(meta: &mut ConstraintSystem<F>) -> Self {
+        Self {
+            byte_col: meta.lookup_table_column(),
+            is_padding_col: meta.lookup_table_column(),
+            is_padding_start_col: meta.lookup_table_column(),
+            is_padding_end_col: meta.lookup_table_column(),
+            _marker: PhantomData,
+        }
+    }
+
+    pub(crate) fn load(&self, layouter: &mut impl Layouter<F>) -> Result<(), Error> {
+        const K: u64 = 256;
+        layouter.assign_table(
+            // Only particular combinations are allowed, 
+            // (is_padding_col, is_padding_start_col, is_padding_end_col, byte_col)
+            // 0 | 0 | 0 | [all 256 possible byte values]
+            // 1 | 1 | 0 | 128 // start-exclusive
+            // 1 | 1 | 1 | 129 // start-end
+            // 1 | 0 | 0 | 0 // mid
+            // 1 | 0 | 1 | 1 // end-exclusive
+
+            || "byte-is_pad-is_pad_start-is_pad_end combination table",
+            |mut table| {
+                for i in 0..K {
+                    table.assign_cell(|| "byte_col_[i=0:K-1]", self.byte_col, i as usize, || Ok(F::from(i)))?;
+                    table.assign_cell(|| "is_padding_col_[i=0:K-1]", self.is_padding_col, i as usize, || Ok(F::from(0)))?;
+                    table.assign_cell(|| "is_padding_start_col_[i=0:K-1]", self.is_padding_start_col, i as usize, || Ok(F::from(0)))?;
+                    table.assign_cell(|| "is_padding_end_col_[i=0:K-1]", self.is_padding_end_col, i as usize, || Ok(F::from(0)))?;
+                }
+                
+                // the middle of the padding case
+                table.assign_cell(|| "byte_col_[i=K]", self.byte_col, (K) as usize, || Ok(F::from(0)))?;
+                table.assign_cell(|| "is_padding_col_[i=K]", self.is_padding_col, (K) as usize, || Ok(F::from(1)))?;
+                table.assign_cell(|| "is_padding_start_col_[i=K]", self.is_padding_start_col, (K) as usize, || Ok(F::from(0)))?;
+                table.assign_cell(|| "is_padding_end_col_[i=K]", self.is_padding_end_col, (K) as usize, || Ok(F::from(0)))?;
```

Comment:
Small nitpick, but normally you'd either create a function that takes the input for the different columns on a single row so the column stuff code doesn't have to be repeated each time for the different row types. Or alternatively you create the input data arrays and use that to create a single `for` loop that uses the array data to set the values on each row in the loop.